k8s/client.go

```diff
@@ -121,6 +123,20 @@ func (c *Client) ClusterName() (name string) {
 	return
 }
 
+func (c *Client) GetAPIServerHostAndPort() (string, string) {
+	if context, ok := c.RawConfig.Contexts[c.ContextName()]; ok {
+		addr := c.RawConfig.Clusters[context.Cluster].Server
```

Comment:
I'm wondering whether this returns a ClusterIP or a real endpoint IP when kube-proxy is not running. Could you check? This is important, because if it returns the ClusterIP, then Cilium won't be able to connect to the kube-apiserver, as no ClusterIP => endpoint IP mapping exists in the Cilium's BPF LB (the chicken and egg problem).

Reply:
Your point makes sense. I checked without `kube-proxy` running(without kube-proxy add-ons), and it always returns the IP of the VM. Another thing, I noticed, It took the server value available in `/etc/kubernetes/admin.conf`.  

Already, We use the host returned by this value to predict the kind of clusters. [AKS/EKS](https://github.com/cilium/cilium-cli/blob/master/k8s/client.go#L523).  Do You think anything else we can check to confirm this?  

Further, for the command `cilium context`, We use this value and print it as `Host`.

```
root@kcp:/home/ubuntu/cilium-cli# ./cilium context
Context: kubernetes-admin@kubernetes
Cluster: kubernetes
Auth: kubernetes-admin
Host: https://192.168.64.1:6443
```

Reply:
Thanks for checking. I think we are good with this.